src/main/scala/is/hail/expr/FunctionRegistry.scala

```diff
@@ -1666,6 +1685,25 @@ object FunctionRegistry {
     "k" -> "Key name to query."
   )(dictHr(TTHr, TUHr), TTHr, boolHr)
 
+  registerLambda("uniroot", { (f: (Any) => Any, min: Double, max: Double) =>
+    val r = uniroot({ (x: Double) =>
+      val y = f(x)
+      if (y == null)
```

Comment:
The R docs for uniroot say:

 > Either interval or both lower and upper must be specified: the upper endpoint must be strictly larger than the lower endpoint. The function values at the endpoints must be of opposite signs (or zero), for extendInt="no", the default. Otherwise, if extendInt="yes", the interval is extended on both sides, in search of a sign change, i.e., until the search interval [l,u] satisfies f(l) * f(u) <= 0.

I assume you implemented extendInt="no"?  If so, I will update the docs to check the sign at the endpoints are opposite.

 > Convergence is declared either if f(x) == 0 or the change in x for one step of the algorithm is less than tol (plus an allowance for representation error in x).

Your code does the same, right?

 > If the algorithm does not converge in maxiter steps, a warning is printed and the current approximation is returned.

We don't have a maxiter argument.  How do you handle this?

Reply:
1. Yes, implemented the default extendInt="no" because that's what fisher.test called.
2. Exactly. `val toleranceActual = 2 * epsilon * math.abs(b) + tolerance / 2`
3. The max number of iterations is hard coded at 1000, which is the default.

I implemented the bare minimum required for FET to work and give the same exact answer as R. I tested the code by running 100,000+ iterations of FET and comparing with R. I did not explicitly test against the uniroot function in R with a variety of function inputs. If we need to more rigorously test and take a second look at the code, can you make an issue? We probably could also make the code faster as I used a lot of the scala functionality like `zipWithIndex` etc.